### ISSUE TITLE
微调qwen3vl8B, 除第一个batch，后面的loss均为0是为什么，数据均按对应格式构造的

### ISSUE THREAD

**Issue by lingfengren** [QwenLM/Qwen3-VL#1658]
使用zero2会这样
```
{'loss': 1.502, 'grad_norm': 1.4142135381698608, 'learning_rate': 0.0, 'epoch': 0.0}
{'loss': 0.0, 'grad_norm': 1.4142135381698608, 'learning_rate': 8.333333333333333e-07, 'epoch': 0.01}
{'loss': 0.0, 'grad_norm': 1.4142135381698608, 'learning_rate': 1.6666666666666667e-06, 'epoch': 0.01}
{'loss': 0.0, 'grad_norm': 1.4142135381698608, 'learning_rate': 2.5e-06, 'epoch': 0.01}
  1%|          | 4/389 [01:45<2:09:22, 20.16s/it]
```

但是调整到zero3就正常了
```
{'loss': 1.4369, 'grad_norm': 41.78952314827526, 'learning_rate': 0.0, 'epoch': 0.0}
{'loss': 1.4625, 'grad_norm': 41.058763466019315, 'learning_rate': 8.333333333333333e-07, 'epoch': 0.01}
{'loss': 1.4579, 'grad_norm': 37.240542738549294, 'learning_rate': 1.6666666666666667e-06, 'epoch': 0.01}
{'loss': 1.3806, 'grad_norm': 30.87280681345312, 'learning_rate': 2.5e-06, 'epoch': 0.01}
{'loss': 1.0932, 'grad_norm': 20.14320310466894, 'learning_rate': 3.3333333333333333e-06, 'epoch': 0.01}
{'loss': 1.0852, 'grad_norm': 12.257438997039708, 'learning_rate': 4.166666666666667e-06, 'epoch': 0.02}
{'loss': 0.9613, 'grad_norm': 10.62834183883574, 'learning_rate': 5e-06, 'epoch': 0.02}
  2%|▏         | 7/389 [02:32<1:40:52, 15.84s/it]
```

环境如下
```
Package                           Version
--------------------------------- -------------
absl-py                           2.3.1
accelerate                        1.10.1
addict                            2.4.0
aiofiles                          24.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.0
aiosignal                         1.4.0
aliyun-python-sdk-core            2.16.0
aliyun-python-sdk-kms             2.16.5
annotated-types                   0.7.0
anyio                             4.11.0
astor                             0.8.1
async-timeout                     5.0.1
attrdict                          2.0.1
attrs                             25.4.0
av                                15.1.0
binpacking                        1.5.2
blake3                            1.0.8
Brotli                            1.1.0
cachetools                        6.2.1
cbor2                             5.7.0
certifi                           2025.10.5
cffi                              2.0.0
charset-normalizer                3.4.3
click                             8.2.1
cloudpickle                       3.1.1
compressed-tensors                0.11.0
contourpy                         1.3.2
cpm-kernels                       1.0.11
crcmod                            1.7
cryptography                      46.0.3
cupy-cuda12x                      13.6.0
cycler                            0.12.1
dacite                            1.9.2
datasets                          3.6.0
deepspeed                         0.17.1
depyf                             0.19.0
dill                              0.3.8
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
einops                            0.8.1
email-validator                   2.3.0
exceptiongroup                    1.3.0
fastapi                           0.118.2
fastapi-cli                       0.0.13
fastapi-cloud-cli                 0.3.1
fastrlock                         0.8.3
ffmpy                             0.6.2
filelock                          3.20.0
flash-attn                        2.8.3
fonttools                         4.60.1
frozendict                        2.4.6
frozenlist                        1.8.0
fsspec                            2025.3.0
future                            1.0.0
gguf                              0.17.1
gitdb                             4.0.12
GitPython                         3.1.45
gradio                            5.49.1
gradio_client                     1.13.3
groovy                            0.1.2
grpcio                            1.75.1
h11                               0.16.0
hf-xet                            1.1.10
hjson                             3.1.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.35.3
idna                              3.10
importlib_metadata                8.7.0
interegular                       0.3.3
jieba                             0.42.1
Jinja2                            3.1.6
jiter                             0.11.1
jmespath                          0.10.0
joblib                            1.5.2
json_repair                       0.52.1
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
kiwisolver                        1.4.9
lark                              1.2.2
llguidance                        0.7.30
llvmlite                          0.44.0
lm-format-enforcer                0.11.3
Markdown                          3.9
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
matplotlib                        3.10.7
mdurl                             0.1.2
mistral_common                    1.8.5
modelscope                        1.30.0
mpmath                            1.3.0
ms_swift                          3.9.1
msgpack                           1.1.2
msgspec                           0.19.0
multidict                         6.7.0
multiprocess                      0.70.16
networkx                          3.4.2
ninja                             1.13.0
nltk                              3.9.2
numba                             0.61.2
numpy                             2.2.6
nvidia-cublas-cu12                12.8.4.1
nvidia-cuda-cupti-cu12            12.8.90
nvidia-cuda-nvrtc-cu12            12.8.93
nvidia-cuda-runtime-cu12          12.8.90
nvidia-cudnn-cu12                 9.10.2.21
nvidia-cufft-cu12                 11.3.3.83
nvidia-cufile-cu12                1.13.1.3
nvidia-curand-cu12                10.3.9.90
nvidia-cusolver-cu12              11.7.3.90
nvidia-cusparse-cu12              12.5.8.93
nvidia-cusparselt-cu12            0.7.1
nvidia-ml-py                      13.580.82
nvidia-nccl-cu12                  2.27.3
nvidia-nvjitlink-cu12             12.8.93
nvidia-nvtx-cu12                  12.8.90
openai                            2.5.0
openai-harmony                    0.0.4
opencv-python-headless            4.12.0.88
orjson                            3.11.3
oss2                              2.19.1
outlines_core                     0.2.11
packaging                         25.0
pandas                            2.3.3
partial-json-parser               0.2.1.1.post6
peft                              0.17.1
pillow                            11.3.0
pip                               25.2
platformdirs                      4.5.0
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.4.1
protobuf                          6.33.0
psutil                            7.1.0
py-cpuinfo                        9.0.0
pyarrow                           21.0.0
pybase64                          1.4.2
pycountry                         24.6.1
pycparser                         2.23
pycryptodome                      3.23.0
pydantic                          2.11.10
pydantic_core                     2.33.2
pydantic-extra-types              2.10.6
pydub                             0.25.1
Pygments                          2.19.2
pyparsing                         3.2.5
python-dateutil                   2.9.0.post0
python-dotenv                     1.1.1
python-json-logger                4.0.0
python-multipart                  0.0.20
pytz                              2025.2
PyYAML                            6.0.3
pyzmq                             27.1.0
qwen-vl-utils                     0.0.14
ray                               2.50.1
referencing                       0.37.0
regex                             2025.9.18
requests                          2.32.5
rich                              14.1.0
rich-toolkit                      0.15.1
rignore                           0.7.1
rouge                             1.0.1
rpds-py                           0.27.1
ruff                              0.14.0
safehttpx                         0.1.6
safetensors                       0.6.2
scipy                             1.15.3
semantic-version                  2.10.0
sentencepiece                     0.2.1
sentry-sdk                        2.42.0
setproctitle                      1.3.7
setuptools                        78.1.1
shellingham                       1.5.4
simplejson                        3.20.2
six                               1.17.0
smmap                             5.0.2
sniffio                           1.3.1
sortedcontainers                  2.4.0
soundfile                         0.13.1
soxr                              1.0.0
starlette                         0.48.0
sympy                             1.14.0
tensorboard                       2.20.0
tensorboard-data-server           0.7.2
tiktoken                          0.12.0
tokenizers                        0.22.1
tomlkit                           0.13.3
torch                             2.8.0
torchaudio                        2.8.0
torchcodec                        0.7.0
torchvision                       0.23.0
tqdm                              4.67.1
transformers                      4.57.1
transformers-stream-generator     0.0.5
triton                            3.4.0
trl                               0.23.1
typer                             0.19.2
typer-slim                        0.19.2
typing_extensions                 4.15.0
typing-inspection                 0.4.2
tzdata                            2025.2
urllib3                           2.5.0
uvicorn                           0.37.0
uvloop                            0.22.1
vllm                              0.11.0
wandb                             0.22.2
watchfiles                        1.1.1
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.45.1
xformers                          0.0.32.post1
xgrammar                          0.1.25
xxhash                            3.6.0
yarl                              1.22.0
zipp                              3.23.0
zstandard                         0.25.0
```

**Comment by litengfeigithub** [comment]
请问您这边微调8B模型所需要的显存为多大？

**Comment by lingfengren** [comment]
> 请问您这边微调8B模型所需要的显存为多大？

八卡机h20，跑zero2时，单卡占用
`52171MiB / 97871MiB`

**Comment by yaoqi-isee** [comment]
mark，我也是这样

**Comment by Bendywater** [comment]
> > 请问您这边微调8B模型所需要的显存为多大？
> 
> 八卡机h20，跑zero2时，单卡占用 `52171MiB / 97871MiB`

啊我也是请问解决了吗😭

**Comment by John-Ge** [comment]
我们也遇到这个情况，目前还没找到原因，我这边 zero2 8B 是可以正常训的，这个问题你们能稳定复现吗

**Comment by lingfengren** [comment]
> 我们也遇到这个情况，目前还没找到原因，我这边 zero2 8B 是可以正常训的，这个问题你们能稳定复现吗

目前使用zero2一直是这个问题，切换到zero3 loss才正常

**Comment by John-Ge** [comment]
我们在查找一下问题

**Comment by yzxyzh** [comment]
> 我们在查找一下问题

我们也发现了一样的问题。我们的配置与题主一样，H20的集群。在微调时发现使用zero2的模型，会出现loss不稳定训练崩溃，现象类似题主的loss突然归0，不过我们的grad norm也有比较异常的表现，会出现较大的波动最后到nan。并且这种现象是可复现的，就是每次启动训练，哪怕使用不同的集群，使用不同的卡数量（16，24，32我都测试过），都会出现错误。但是只要把zero2改成zero3，则一切都正常了。并且似乎只有qwen3-vl-8b会这样，4b则不会。

**Comment by yufeixue-ai** [comment]
使用zero3训练qwen3-vl-2b模型也会出现这样的问题，蹲

**Comment by PeterLuo2003** [comment]
一样的问题，8卡A100使用zero2训练qwen3-vl-2b也会第一个Batch的loss非0，后边全是0，甚至梯度也是和@lingfengren一样的。但是我使用图片和视频的混合数据集无法使用zero3进行训练。

**Comment by YHPeter** [comment]
问题比较奇怪 应该不是qwen3-vl的问题 我在三个不同的集群上面 都是H100 基本上一样的环境 两个有一样的现象 很神奇
Gemma2, Gemma3, Qwen3, Qwen2都一样 
{'loss': 14.6752, 'grad_norm': 1.4142135381698608, 'learning_rate': 0.0, 'epoch': 0.0}
其中一个把FA3->FA2就好了 还有个实在无法解决 均是Z1/Z2

**Comment by PeterLuo2003** [comment]
将zero2.json或zero3.json中的`"overlap_comm": true`改为`"overlap_comm": false`,问题解决

**Comment by Veason-silverbullet** [comment]
The same. This issue occurs occasionally for multi-node training for Qwen-VL-2.5....
